include/dali/core/tensor_shape.h

```diff
@@ -1049,6 +1050,34 @@ TensorListShape<ndim> uniform_list_shape(int num_samples, std::initializer_list<
   return TensorListShape<ndim>::make_uniform(num_samples, sample_shape);
 }
 
+/**
+ * @brief Merges a dimension with the next one
+ *
+ * @details The output shape has two specified dimension collapsed with the next one
+ * [2, 3, 4, 5]
+ * after a call to collapse_dim(tensor, 1) would have a shape:
+ * [2, 12, 5]
+ *
+ * @param dim_idx - the dimension to drop; must be >= 0 and < shape.size() - 1
+ * @param rv - input TensorView
+ * @remarks The `dim_idx` must not be the innermost dimension or the result is undefined.
```

Comment:
This remark should say that it would just assert.

Reply:
It's only true in debug build.